tyrian/js/src/main/scala/tyrian/Sub.scala

```diff
@@ -175,28 +175,20 @@ object Sub:
 
   /** A subscription that produces a `msg` after a `duration`. */
   def timeout[F[_]: Sync, Msg](duration: FiniteDuration, msg: Msg, id: String): Sub[F, Msg] =
-    val acquire: F[(Either[Throwable, Msg] => Unit) => Int] =
-      Sync[F].delay { (callback: Either[Throwable, Msg] => Unit) =>
+    val acquire: (Either[Throwable, Msg] => Unit) => Int =
+      (callback: Either[Throwable, Msg] => Unit) =>
         dom.window.setTimeout(
           Functions.fun0(() => callback(Right(msg))),
           duration.toMillis.toDouble
         )
```

Comment:
I'll take another look at this in the morning when I'm a bit fresher, but my only slight concern here would be that we lose error handling. In practice, I don't think there are any errors to handle though... :thinking:

Reply:
So, the `Sync[F].delay` covers the creation of the function, but it does not cover an execution of the body of the function.